pyat/at.c

```diff
@@ -289,6 +291,15 @@ void set_energy_particle(PyObject *lattice, PyObject *energy,
                 param->charge = PyFloat_AsDouble(pcharge);
                 Py_DECREF(pcharge);
             }
+            if (bcurrent != NULL) {
+                param->beam_current = PyFloat_AsDouble(bcurrent);
+                Py_DECREF(bcurrent);
+            }
+            if (bcurrents != NULL) {
+                param->bunch_currents = PyArray_DATA(bcurrents);
+                param->nbunch = PyArray_SIZE(bcurrents);
+                Py_DECREF(bcurrents);
```

Comment:
@lfarv, this seems to work but I have a doubt, should I DECREF here? Isn't the pointer now pointing to an unreferenced array? In this case why is this working...?